client/src/utils/file.js

```diff
@@ -26,7 +29,23 @@ module.exports = {
         }
         break;
       default:
-        throw new Error(file.type + ' is not a supported file type. Only, .jpeg, .png, .gif, and .mp4 files are currently supported.');
+        // Handle STL check (file.type is empty for STLs and file.type cannot be reassigned)
+        file.isStl = false;
+        return new Promise((resolve, reject) => {
+          const loader = new FileReader();
```

Comment:
This entangles synchronous validation with a threejs loader that uses async events.
Being a synchronous method, this shouldn't wrap with a promise either.